src/main/java/mServer/crawler/sender/orf/tasks/AbstractRecursivConverterTask.java

```diff
@@ -0,0 +1,97 @@
+package mServer.crawler.sender.orf.tasks;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.RecursiveTask;
+import mServer.crawler.sender.MediathekReader;
+
+/**
+ * This task is based on {@link RecursiveTask} and takes a
+ * {@link ConcurrentLinkedQueue} of {@link D}. It splits the elements to process
+ * on instances of it self based on the crawler configuration and calls the
+ * {@link this#processElement(Object))} for each.
+ *
+ * @author Nicklas Wiegandt (Nicklas2751)<br>
+ * <b>Mail:</b> nicklas@wiegandt.eu<br>
+ * <b>Jabber:</b> nicklas2751@elaon.de<br>
+ * <b>Riot.im:</b> nicklas2751:matrix.elaon.de<br>
+ *
+ * @param <T> The type of objects which will be created from this task.
+ * @param <D> A result objects type.
+ */
+public abstract class AbstractRecursivConverterTask<T, D> extends RecursiveTask<Set<T>> {
+
+  private static final long serialVersionUID = 8416254950859957820L;
+
+  private final ConcurrentLinkedQueue<D> elementsToProcess;
+
+  /**
+   * The crawler which this task is for.
+   */
+  protected transient MediathekReader crawler;
+
+  /**
+   * The set of results. This set will be returned at the end of the task.
+   */
+  protected transient Set<T> taskResults;
+
+  public AbstractRecursivConverterTask(final MediathekReader aCrawler,
+          final ConcurrentLinkedQueue<D> aUrlToCrawlDTOs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDTOs" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)